cpp/src/arrow/adapters/orc/util.cc

```diff
@@ -18,6 +18,7 @@
 #include "arrow/adapters/orc/util.h"
 
 #include <cmath>
+#include <sstream>
```

Comment:
Why do we need this?

Reply:
This is used in `GetOrcMajorVersion()`, so it should be #included here by IWYU. I found it because compilation failed at one point (compilation can spuriously succeed when standard headers include more definitions than the standard guarantees; for example, inclusion of only `<string>` or `<iostream>` might fully define `std::stringstream` rather than only forward declaring it)